### PR TITLE
Test de-flaking & bug fixes

### PR DESCRIPTION
**Pull request by JoshVanL**
This PR removes the _majority_ of flakes in our integration and unit tests. The changes include:
- Increasing retry timeouts for actor related integration test clients.
- Allowing for gRPC server stopped errors in [placement_test.go](https://github.com/dapr/dapr/pull/7377/files#diff-98b51d9418a5953b74f395a5e46696e7a0e704c79f3e3291f60ac9a52445e225R57).
- Fixing a race condition in the [binding send unit test](https://github.com/dapr/dapr/pull/7377/files#diff-af0c26bb14fb64768899657678c05a8034faa62648e7a12d02e59483cdfcd227R69).
- Fixing a panic race condition in the sentry JWK validator but ensuring the cache is warm before [validating tokens](https://github.com/dapr/dapr/pull/7377/files#diff-8c2cfced615ebe60378491598d389de4a00a2dbdf2a8bf56591a6bb19cc0fe30R59).
- Fixing race condition in pluggable component int tests by ensuring the pluggable gRPC server is [initialized before running](https://github.com/dapr/dapr/pull/7377/files#diff-867af9c8bbca0622fa8668268f729bb9a795a02e399ebd8af7280ecaafefb179R95).
- Preventing an test assertion panic if the metadata server is not ready when called in [int tests](https://github.com/dapr/dapr/pull/7377/files#diff-3f912876ef7bbebb3062480ac885ec4d1bd36efa0b78dea0c09e423a2c303cf9R43).
- Cosmetic: Integration tests are now run in alphabetical order by name, and the [execution time is outputted](https://github.com/dapr/dapr/pull/7377/files#diff-3ad53d769bbbfc0927a6389cf77e37edbb556e733e7a3de84bb640bda96b3740R48).
- Fix panic race in [actors/healthz/initerror](https://github.com/dapr/dapr/pull/7377/files#diff-b9a7b5de88a3c267928c650d609366c545ff0a1f0067c645b8c9e2b6d1f37d51R63).
- Fix panic race in [actors/http/ttl](https://github.com/dapr/dapr/pull/7377/files#diff-cdb3b91140322033d1972262ba116140af1dcda59d1a17e4a807c1b27e85c2ccR71).
- Fix test assertion panic in [reminders/basic](https://github.com/dapr/dapr/pull/7377/files#diff-9b64d52deae20723371b4592e9d5ef8f60274bb16d13f39bceea80c8fac3a775R92).
- Fix test assertion panic in [reminders/serialisation](https://github.com/dapr/dapr/pull/7377/files#diff-8f5590a98db6dc6510af1e39beef2dbb706b0a3ba88bf6da57245d21784d62b8R36).
- Update all metadata server int test clients to expect the newly always available workflow component, introduced in https://github.com/dapr/dapr/pull/7283
- JSON marshal all file name strings to they are compatible with [windows runners](https://github.com/dapr/dapr/pull/7377/files#diff-fbb1ce21d65713155c474ba284aee9f3a8da87b32e2ee112534e53781e688e10R84).
- Skip pluggable tests on windows as they rely on [unix sockets which are not available yet for windows](https://github.com/dapr/dapr/pull/7377/files#diff-45140521fed3b5d87f7e7de90010983c0256b37704ab67a4c0dce346d238da55R60).
- Make placement apiLevel int tests [retry connection with placement](https://github.com/dapr/dapr/pull/7377/files#diff-8c06ed9c48a13e82d8cee89ef1137c429d34e907dfc4ac2c57028f9d1c41705cR64).
- Updates kit to use [PR](https://github.com/dapr/kit/pull/82) to fix deadlock in fswatcher.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -466,6 +466,8 @@ replace (
 // check for retracted versions: go list -mod=mod -f '{{if .Retracted}}{{.}}{{end}}' -u -m all
 replace github.com/microcosm-cc/bluemonday => github.com/microcosm-cc/bluemonday v1.0.24
 
+replace github.com/dapr/kit => github.com/joshvanl/kit v0.0.0-20240114211428-c517e2d0d025
```
Comment (artursouza) [inline go.mod:469]: Use the main version, please.

**Review by artursouza — APPROVED** [review]

**Review by artursouza — APPROVED** [review]

**Review by mukundansundar — DISMISSED** [review]

**Review by elena-kolevska — APPROVED** [review]

**Review by yaron2 — APPROVED** [review]
Great PR

**Pull request merged**